core/src/test/java/com/uber/m3/tally/ScopeImplTest.java

```diff
@@ -229,32 +229,32 @@ public void snapshot() {
 
         Map<ScopeKey, CounterSnapshot> counters = snapshot.counters();
         assertEquals(1, counters.size());
-        CounterSnapshot counterSnapshotActual = counters.get(ScopeImpl.keyForPrefixedStringMap("snapshot-counter", null));
+        CounterSnapshot counterSnapshotActual = counters.get(ScopeImpl.keyForPrefixedStringMap("snapshot-counter", ImmutableMap.EMPTY));
         assertEquals("snapshot-counter", counterSnapshotActual.name());
-        assertEquals(null, counterSnapshotActual.tags());
+        assertEquals(ImmutableMap.EMPTY, counterSnapshotActual.tags());
 
         Map<ScopeKey, GaugeSnapshot> gauges = snapshot.gauges();
         assertEquals(3, gauges.size());
-        GaugeSnapshot gaugeSnapshotActual = gauges.get(ScopeImpl.keyForPrefixedStringMap("snapshot-gauge", null));
+        GaugeSnapshot gaugeSnapshotActual = gauges.get(ScopeImpl.keyForPrefixedStringMap("snapshot-gauge", ImmutableMap.EMPTY));
         assertEquals("snapshot-gauge", gaugeSnapshotActual.name());
-        assertEquals(null, gaugeSnapshotActual.tags());
+        assertEquals(ImmutableMap.EMPTY, gaugeSnapshotActual.tags());
         assertEquals(120, gaugeSnapshotActual.value(), EPSILON);
 
-        GaugeSnapshot gaugeSnapshot2Actual = gauges.get(ScopeImpl.keyForPrefixedStringMap("snapshot-gauge2", null));
+        GaugeSnapshot gaugeSnapshot2Actual = gauges.get(ScopeImpl.keyForPrefixedStringMap("snapshot-gauge2", ImmutableMap.EMPTY));
```

Comment:
should we change `keyForPrefixedStringMap` implementation so that if null is provided we substitute it with an empty Map?